website/preprints/tasks.py

```diff
@@ -113,11 +113,8 @@ def format_preprint(preprint, share_type, old_subjects=None):
         'title': preprint.node.title,
         'description': preprint.node.description or '',
         'is_deleted': (
-            not preprint.is_published or
-            not preprint.node.is_public or
-            preprint.node.is_preprint_orphan or
-            preprint.node.tags.filter(name='qatest').exists() or
-            preprint.node.is_deleted
+            not preprint._verified_publishable or
```

Comment:
It does check `preprint.node.is_preprint`, which seems like it covers those?
```python
    @property
    def is_preprint(self):
        # TODO: This is a temporary implementation.
        if not self.preprint_file_id or not self.is_public:
            return False
        if self.preprint_file.node_id == self.id:
            return self.has_published_preprint
        else:
            self._is_preprint_orphan = True
            return False
```
I think it might be more thorough than the old code? Downside is `has_published_preprint` makes a redundant query, but I figure it'll have to be reworked in the node/preprint divorce anyway, and it seems worth it to have all the "should this preprint be publicly visible?" logic in one place.